### PR TITLE
CLDR-15857 simplify person names code

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-15857

Simplify the code by replacing ParameterMatcher by FormatParameters.
ROUGH CUT for now.

- [ ] This PR completes the ticket.

### CODE REVIEW

**Comment by macchiati** [comment]
Hi, this is a very first cut at simplifying the code.

**Comment by macchiati** [comment]
The tests pass, and I added a test to check that we can format all the sample names. But it still needs cleanup.
- remove commented-out code
- mark methods that are only public for testing
- add the API we discussed (eg to option to uppercase surname for surnameFirst if the locale is givenFirst
- other cleanup

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java [hunk 1]
```diff
@@ -981,263 +1041,273 @@ private <T> void addToLabel(T item, StringBuilder sb) {
...
+//            // now replace any "complete" items by empty.
+//            Set<ParameterMatcher> result2 = new TreeSet<>();
+//            for (ParameterMatcher item1 : result) {
+//                result2.add(item1.slim());
+//            }
+//            return ImmutableSet.copyOf(result2);
+//        }
+//
+//    }
+
```
Comment (richgillam) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java:1302]: Do you need to keep all this commented-out code in here?  If you can just get it out of a previous revision in git, that'd be cleaner.  (Or is this just still here until you're completely done with this simplification effort, and THEN you'll delete it?)

**Review by richgillam — APPROVED** [review]
Didn't read it EXTREMELY closely, but it all looks pretty straightforward.

**Comment by macchiati** [comment]
Yes, I left it in just during development; will delete.

**Review by richgillam — APPROVED** [review]
Looks good!

**Pull request merged**